index.ts

```diff
@@ -44,52 +53,96 @@ const buildMessageString = ({
 [View Results](${url})
 `;
 
+const uploadBuild = async (
+  url: URL,
+  apiKey: string,
+  buildPath: string
+): Promise<IBuildUploadResponse> => {
+  if (!existsSync(buildPath)) {
+    throw new Error('Build file not found');
+  }
+
+  const fileName = path.basename(buildPath);
+  const fileData = await readFile(buildPath);
+  const formData = new FormData();
+  formData.append('file', fileData, {
+    filename: fileName,
+    filepath: buildPath,
+  });
+
+  const buildUpload = await fetch(`${url}apps/builds`, {
+    method: 'POST',
+    body: formData,
+    headers: {
+      'X-App-Api-Key': apiKey,
+      'User-Agent': 'moropo-github-action',
+    },
+  });
+
+  const responseJson = await buildUpload.json();
+
+  if (!buildUpload.ok) {
+    throw new Error(`Failed to upload build: ${JSON.stringify(responseJson)}`);
+  }
+  console.info('Successfully uploaded build.');
+
+  return responseJson;
+};
+
 const run = async (): Promise<void> => {
   try {
-    let expoReleaseChannel: string | null = core.getInput(
-      "expo_release_channel"
-    );
+    let expoReleaseChannel: string | null = getInput('expo_release_channel');
     if (!expoReleaseChannel?.length) {
       expoReleaseChannel = null;
     }
-    const testRunId = core.getInput("scheduled_test_id");
-    const moropoApiKey = core.getInput("api_key");
-    const githubToken = core.getInput("github_token");
-
-    const headers: Record<string, string> = {
-      "Content-Type": "application/json",
-    };
-
-    if (moropoApiKey) {
-      headers["x-moropo-api-key"] = moropoApiKey;
+    const scheduledTestRunId = getInput('scheduled_test_id');
+    const apiKey = getInput('api_key');
+    const githubToken = getInput('github_token');
+    const buildPath = getInput('build_path');
+    const moropoUrl = new URL(getInput('moropo_url'));
+    const moropoApiUrl = new URL(getInput('moropo_api_url'));
+
+    if (!moropoUrl || !moropoApiUrl || !apiKey || !scheduledTestRunId) {
+      setFailed(
+        'Missing one or more required inputs: moropo_url, moropo_api_url, api_key, scheduled_test_id'
```

Comment:
probably shouldnt include moropo urls in here

Reply:
They're automatically considered present as they are set as defaults in the `action.yml`

Reply:
it would confuse a user if they dont provide apikey or test run id but then get told that moropo url is missing